Benchmarks/Benchmarks/NIOPosixBenchmarks/TCPEchoAsyncChannel.swift

```diff
@@ -0,0 +1,89 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@_spi(AsyncChannel) import NIOCore
+@_spi(AsyncChannel) import NIOPosix
+
+func runTCPEchoAsyncChannel(numberOfWrites: Int, eventLoop: EventLoop) async throws {
+    let serverChannel = try await ServerBootstrap(group: eventLoop)
+        .bind(
+            host: "127.0.0.1",
+            port: 0
+        ) { channel in
+            channel.eventLoop.makeCompletedFuture {
+                return try NIOAsyncChannel(
+                    synchronouslyWrapping: channel,
+                    configuration: .init(
+                        inboundType: ByteBuffer.self,
+                        outboundType: ByteBuffer.self
+                    )
+                )
+            }
+        }
+
+    let clientChannel = try await ClientBootstrap(group: eventLoop)
+        .connect(
+            host: "127.0.0.1",
+            port: serverChannel.channel.localAddress!.port!
+        ) { channel in
+            channel.eventLoop.makeCompletedFuture {
+                return try NIOAsyncChannel(
+                    synchronouslyWrapping: channel,
+                    configuration: .init(
+                        inboundType: ByteBuffer.self,
+                        outboundType: ByteBuffer.self
+                    )
+                )
+            }
+        }
+
+    let bufferSize = 10000
```

Comment:
nit: naming threw me for a sec as I expected a buffer, not a `ByteBuffer`. something like `messageSize` is a more obvious